keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ResourceRepository.kt

```diff
@@ -45,13 +45,20 @@ interface ResourceRepository : PeriodicallyCheckedRepository<Resource<ResourceSp
   fun allResources(callback: (ResourceHeader) -> Unit)
 
   /**
-   * Retrieves a single resource by its unique [id].
+   * Retrieves a single resource by its unique [id]. If the resource is of an obsolete kind, the
+   * result is automatically migrated to the latest version of that kind.
    *
    * @return The resource represented by [id] or `null` if [id] is unknown.
    * @throws NoSuchResourceException if [id] does not map to a resource in the repository.
    */
   fun get(id: String): Resource<ResourceSpec>
 
```

Comment:
Can you add your comment about how this method should be used here as well?

Reply:
Is that better?